plugins/modules/net_tools/nmcli.py

```diff
@@ -923,7 +923,7 @@
                 description: This defines the service type of connection.
                 type: str
                 required: true
-                choices: [ pptp, l2tp ]
+                choices: [ org.freedesktop.NetworkManager.pptp, org.freedesktop.NetworkManager.l2tp ]
```

Comment:
This is a breaking change, since previous valid values for this option are now no longer allowed.

You might want to add the two new modules, and write code which converts the old values to the new values. I assume that nmcli internally converted the short to the long version? Also, is this true for all nmcli versions, or are there some where the short name is actually required?

Reply:
Hmm, I wonder if changing this to accept any string is a good idea or not. `nmcli.py` does not treat pptp or l2tp specially. I believe this can already work for other kinds of VPNs

> You might want to add the two new modules, and write code which converts the old values to the new values.

Not sure what this means. Is there a module with such a pattern?

Reply:
Sorry, I meant "add the two new choices"...

About accepting any string: I guess that's also fine, if there are more possible choices. In that case the user has to make sure themself to pass in a value that is idempotent.

Reply:
> About accepting any string: I guess that's also fine, if there are more possible choices.

Thanks, I took this way in the latest commit. As a side note, I can setup OpenVPN with current `nmcli.py`:

```YAML
- name: configure OpenVPN
  community.general.nmcli:
    conn_name: '{{ item.name }}'
    type: vpn
    vpn:
      service-type: org.freedesktop.NetworkManager.openvpn
      remote: '{{ lookup("community.general.dig", item.gateway) }}'
      username: '{{ item.username }}'
      connection-type: password
      password-flags: 0
      ca: /etc/ssl/cert.pem
      cipher: '{{ item.cipher | default(omit) }}'
    autoconnect: false
    state: present
  loop: '{{ openvpn }}'
```
An issue is that the openvpn plugin uses different keys for the gateway address and the username. To make things worse, apparently neither are documented.

Reply:
Oh the fun... Not having `choices` here seems like the best solution.